templates/shift-scheduling/schema.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/nextmv-io/sdk/mip"
+)
+
+// output holds the output data of the solution.
+type output struct {
+	AssignedShifts        []outputAssignment `json:"assigned_shifts"`
+	NumberAssignedWorkers int                `json:"number_assigned_workers"`
+}
+
+// options holds custom configuration data.
+type options struct {
+	OverSupplyPenalty  float64       `json:"over_supply_penalty" default:"1000" usage:"penalty for over-supplying a demand"`
+	UnderSupplyPenalty float64       `json:"under_supply_penalty" default:"500" usage:"penalty for over-supplying a demand"`
+	MaxHoursPerDay     time.Duration `json:"max_hours_per_day" default:"10h" usage:"maximum number of hours per day"`
+	MaxHoursPerWeek    time.Duration `json:"max_hours_per_week" default:"40h" usage:"maximum number of hours per week"`
+	MinHoursPerShift   time.Duration `json:"min_hours_per_shift" default:"2h" usage:"minimum number of hours per shift"`
+	MaxHoursPerShift   time.Duration `json:"max_hours_per_shift" default:"8h" usage:"maximum number of hours per shift"`
+	HoursBetweenShifts time.Duration `json:"hours_between_shifts" default:"8h" usage:"minimum number of hours between shifts"`
+	Limits             mip.Limits    `json:"limits,omitempty"`
+}
+
+// input represents a struct definition that can read input.json.
+type input struct {
+	Workers         []worker         `json:"workers"`
+	RequiredWorkers []requiredWorker `json:"required_workers"`
+}
+
+// worker holds worker specific data.
+type worker struct {
+	Availability []availability `json:"availability"`
+	ID           int            `json:"id"`
+}
+
+// availability holds available times for a worker.
+type availability struct {
+	Start customTime `json:"start"`
+	End   customTime `json:"end"`
+}
+
+// requiredWorker holds data about times and number of required workers per time window.
+type requiredWorker struct {
+	RequiredWorkerID int        `json:"required_worker_id,omitempty"`
+	Start            customTime `json:"start"`
+	End              customTime `json:"end"`
+	Count            int        `json:"count"`
+}
+
+// ID returned the RequiredWorker ID.
+func (r requiredWorker) ID() string {
+	return strconv.Itoa(r.RequiredWorkerID)
+}
+
+// customTime represents a time.Time.
+type customTime struct {
```

Comment:
Why do we need a custom time? And if we need it, why not do `type customTime time.Time`?